tests/python/test_multidevice_overlap.py

```diff
@@ -88,3 +92,21 @@ def test_overlap_allgather_matmul_stream_outermost(multidevice_test, benchmark):
 
     # benchmark
     benchmark.pedantic(lambda: fd.execute(ins), rounds=N_ITERATIONS)
+
+
+@pytest.mark.mpi
+def test_overlap_allgather_matmul_stream_outermost_ucc(multidevice_test, benchmark):
+    os.environ["UCC_CL_BASIC_TLS"] = "nccl"
+    exec_overlap_allgather_matmul_stream_outermost(
+        multidevice_test, benchmark, CommunicatorBackend.ucc
+    )
+    # Resetting the cache here is necessary to workaround a bug that would need a proper fix. If not avoiding the cache, there is an issue for the second test that is being run. More specifically, the second time we define the fusion, we hit the cache in https://github.com/NVIDIA/Fuser/blob/6ff60e2a320733a2f49de57007d6bb45000107cd/csrc/python_frontend/fusion_definition.cpp#L95 . Later, when we call _set_device_mesh, we get a "thro out of range" here https://github.com/NVIDIA/Fuser/blob/6ff60e2a320733a2f49de57007d6bb45000107cd/csrc/python_frontend/schedule_bindings.cpp#L60 because the FusionDefinition has not run so it doesn't contain any state.
```

Comment:
@wujingyue here I am working around an issue that would need a proper fix

Reply:
Nice findings! 

This is apparently a bug in FusionCache. A new FusionDefinition attempts to reuse the same FusionSchedules for the same definition and skips [buildFusionIr](https://github.com/NVIDIA/Fuser/blob/4eb5b763915c0e674f8c2917f1d9173fc55a30f0/csrc/python_frontend/fusion_definition.cpp#L108) which is responsible for building fusion states. Later, [`_setup_multidevice_schedule`](https://github.com/NVIDIA/Fuser/blob/4eb5b763915c0e674f8c2917f1d9173fc55a30f0/nvfuser/__init__.py#L287) tries to access non-existent fusion states. 

cc @rdspring1 this is another two-hour debugging that the direct bindings can save me